File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.rb
# Definition for singly-linked list.
# class ListNode
#     attr_accessor :val, :next
#     def initialize(val = 0, _next = nil)
#         @val = val
#         @next = _next
#     end
# end
# @param {ListNode} list1
# @param {ListNode} list2
# @return {ListNode}
def merge_two_lists(list1, list2)
    if !(list1 || list2)
        @head=[]
    elsif list1 && list2 && list1.val <= list2.val
        @head=ListNode.new(list1.val)
        list1=list1.next
    elsif list1 && list2 && list1.val > list2.val
        @head=ListNode.new(list2.val)
        list2=list2.next
    elsif list1
        @head=ListNode.new(list1.val)
        list1=list1.next
    else
        @head=ListNode.new(list2.val)
        list2=list2.next
    end
    final=@head
    @list1=list1
    @list2=list2
    while !@list1.nil? && !@list2.nil?
        if @list1.val==@list2.val
            @head.next=ListNode.new(@list1.val)
            @head.next.next=ListNode.new(@list2.val)
            @head=@head.next.next
            @list1=@list1.next
            @list2=@list2.next
        elsif @list1.val<@list2.val
            @head.next=ListNode.new(@list1.val)
            @head=@head.next
            @list1=@list1.next
        else
            @head.next=ListNode.new(@list2.val)
            @head=@head.next
            @list2=@list2.next
        end
    end
    while !@list1.nil?
            @head.next=ListNode.new(@list1.val)
            @head=@head.next
            @list1=@list1.next
    end
    while !@list2.nil?
            @head.next=ListNode.new(@list2.val)
            @head=@head.next
            @list2=@list2.next
    end
    final
end

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} list1
 * @param {ListNode} list2
 * @return {ListNode}
 */
var mergeTwoLists = function(list1, list2) {
    let list=null
    if(list1&&list2){
        if(list1.val<list2.val){
            list=list1
            list1=list1.next
        }else{
            list=list2
            list2=list2.next
        }
            
    }
    else if(list1)
        return list1
    else
        return list2
    
    let head=list
    while(list1){
        while(list2&&list2.val<=list1.val){
            list.next=list2
            list2=list2.next
            list=list.next
        }
        list.next=list1
        list1=list1.next
        list=list.next
    }
    
    
        while(list2){
            list.next=list2
            list2=list2.next
            list=list.next
        }
    return head
};